### PR TITLE
fix some eslint stuff

### PR DESCRIPTION
**Pull request by danielzsh**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

**Comment by danielzsh** [comment]
@thecodingwizard what do you think of this tsconfig? I have set strict to false but enabled `strictNullChecks` in tsconfig (which is mostly for purposes of vscode intellisense), while disabling it for now under `check-ts-errors` so old code doesn't break. However, the ultimate goal is to eventually enable `strictNullChecks` in `check-ts-errors` as well (that way people aren't confused about why check-ts-errors throws different output than vscode intellisense) I could also try fixing the `strictNullChecks` in this pr itself if you want! (obviously, the ultimate goal is to have all strict family checks enabled, but I think we can just go incrementally for now)

**Comment by SansPapyrus683** [comment]
what's w/ the changed names

**Comment by danielzsh** [comment]
basically eslint thinks any function that starts with `use` is a react hook (which many of them were not)

**Comment by thecodingwizard** [comment]
What you said makes sense to me! I think it's fine if we incrementally fix `strictNullChecks`; agree that we should do it at some point though. Thanks!!

**Comment by thecodingwizard** [comment]
wait, isn't `useUserLang()` etc considered a hook? What's the eslint error that you see?

**Comment by danielzsh** [comment]
> wait, isn't useUserLang() etc considered a hook? What's the eslint error that you see?

Oh oops, I kinda just renamed all the functions that were throwing errors; lmk if there's any other hooks I accidentally renamed! The error was that it was used in `sectionFromLang`, which isn't allowed by eslint since its not a functional component.

File: src/components/MarkdownLayout/TableOfContents/getSuffix.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { useContext } from 'react';
 import MarkdownLayoutContext from '../../../context/MarkdownLayoutContext';
 import { ModuleInfo } from '../../../models/module';
 
-export default function getSuffix() {
+export default function useSuffix() {
   const moduleLayoutInfo = useContext(MarkdownLayoutContext);
```
Comment (thecodingwizard) [inline src/components/MarkdownLayout/TableOfContents/getSuffix.tsx:6]: maybe rename the file to `useSuffix.tsx`?

**Comment by danielzsh** [comment]
> Oh oops, I kinda just renamed all the functions that were throwing errors; lmk if there's any other hooks I accidentally renamed! The error was that it was used in sectionFromLang, which isn't allowed by eslint since its not a functional component.

@thecodingwizard bump? 😅 any other hooks i may have accidentally renamed

File: src/components/Groups/MembersPage/MembersPage.tsx [hunk 2]
```diff
@@ -2,15 +2,15 @@ import { Link } from 'gatsby';
...
+import useMemberInfoForGroup from '../../../hooks/groups/useMemberInfoForGroup';
 import TopNavigationBar from '../../TopNavigationBar/TopNavigationBar';
 import Layout from '../../layout';
 import SEO from '../../seo';
 import MemberDetail from './MemberDetail';
 
 export default function MembersPage(): JSX.Element {
   const activeGroup = useActiveGroup();
-  const memberInfo = getMemberInfoForGroup(activeGroup.groupData);
+  const memberInfo = useMemberInfoForGroup(activeGroup.groupData);
```
Comment (thecodingwizard) [inline src/components/Groups/MembersPage/MembersPage.tsx:13]: nice!

**Comment by thecodingwizard** [comment]
I could be wrong, but aren't all the functions you renamed actually hooks?

**Comment by SansPapyrus683** [comment]
so what happened here

**Comment by danielzsh** [comment]
umm it's minor qol stuff not sure when I'll work on it

**Comment by SansPapyrus683** [comment]
nononono

**Comment by danielzsh** [comment]
@thecodingwizard I think all the hooks should be properly renamed, would you mind taking a look 😅

File: src/components/MarkdownLayout/ModuleHeaders/ModuleHeaders.tsx [hunk 3]
```diff
@@ -37,13 +37,17 @@ export default function ModuleHeaders({
 
   const lang = useUserLangSetting();
   const setLang = useSetUserLangSetting();
-
+  let problemIDs = [] as string[];
```
Comment (thecodingwizard) [inline src/components/MarkdownLayout/ModuleHeaders/ModuleHeaders.tsx:40]:
```suggestion
  let problemIDs: string[] = [];
```

though I guess either is fine

**Review by thecodingwizard — APPROVED** [review]
amazing. thanks for fixing all these issues!

**Pull request merged**